word_cloud_generator/word_cloud_generator.py

```diff
@@ -0,0 +1,254 @@
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :param auth: A HTTPBasicAuth object for authentication (HTTPBasicAuth)
+    :param headers: A request Header (dict)
+    :param host: The host used to create the URL to send the request (string)
+    :returns: A list with the contents of all valid issues
+    """
+    issues_contents = []
+    return_amount = 50
+    issues_amount = 0
+    while return_amount == 50:
+        url = f"{host}/rest/servicedeskapi/servicedesk/6/queue/182/issue?start={issues_amount}"
+
+        json_load = get_response_json(auth, headers, url)
+
+        issues = json_load.get("values")
+
+        for issue in issues:
+            issue_date = datetime.strptime(
+                issue.get("fields").get("created")[:10], "%Y-%m-%d"
+            )
+            if issue_date < datetime.strptime(issue_filter.get("end_date"), "%Y-%m-%d"):
+                return issues_contents
+            if filter_issue(issue, issue_filter, issue_date):
+                issue_contents = issue.get("fields").get("summary")
+                if issue_contents:
+                    issues_contents.append(issue_contents)
+
+        return_amount = json_load.get("size")
+        issues_amount = issues_amount + return_amount
+
+    return issues_contents
+
+
+def filter_issue(issue, issue_filter, issue_date):
+    """
+    Function to check if an issue passes the set filters
+    :param issue: A dict of an issues contents (dict)
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :param issue_date: The date that the issue was created (string)
+    :returns: If the issue passes the filters
+    """
+    if issue.get("fields").get("assignee"):
+        issue_assigned = issue.get("fields").get("assignee").get("displayName")
+        if issue_filter.get("assigned") and issue_assigned != issue_filter.get(
+            "assigned"
+        ):
+            return False
+    else:
+        return False
+    if issue_date > datetime.strptime(issue_filter.get("start_date"), "%Y-%m-%d"):
+        return False
+    return True
+
+
+def generate_word_cloud(issues_contents, issue_filter, word_cloud_output_location):
+    """
+    Function to generate and save a word cloud
+    :param issues_contents: The summary of every valid issue (list)
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :param word_cloud_output_location: The output location for the word cloud to be saved to
+    """
+    matches = re.findall(r"((\w+([.'](?![ \n']))*[-_]*)+)", issues_contents)
+    if matches:
+        issues_contents = " ".join(list(list(zip(*matches))[0]))
+    issues_contents = filter_word_cloud(issue_filter, issues_contents)
+    word_cloud = WordCloud(
+        width=2000,
+        height=1000,
+        min_font_size=25,
+        max_words=10000,
+        background_color="white",
+        collocations=False,
+        regexp=r"\w*\S*",
```

Comment:
worth providing these as kwargs for the generate word cloud?
you could rewrite the function like so:
```
generate_word_cloud(issues_contents, issue_filter, word_cloud_output_location, **kwargs):
    def generate_word_cloud(issues_contents, issue_filter, word_cloud_output_location, **kwargs):
    """
    Function to generate and save a word cloud
    :param issues_contents: The summary of every valid issue (list)
    :param issue_filter: Dict of filters to check the issues against (dict)
    :param word_cloud_output_location: The output location for the word cloud to be saved to
    :param kwargs: A set of kwargs to pass to WordCloud
        - width
        - height
        - min_font_size
        - max_words
    """
   ...
   
    word_cloud = WordCloud(
        width=kwargs["width"],
        height=kwargs["height"],
        min_font_size=kwargs["min_font_size"],
        max_words=kwargs["max_words"],
        background_color="white",
        collocations=False,
        regexp=r"\w*\S*",
    )
    ...
```